sonaranalyzer-dotnet/rspec/cs/S4564_c#.html

```diff
@@ -0,0 +1,36 @@
+<p>ASP.NET is coming with a feature looking at HTTP requests to prevent potentially dangerous content to perform a cross-site scripting (XSS) attack.
+There is no reason to disable this mechanism even if it should not be the only one in place to prevent XSS attacks.</p>
+<p>This rule raises an issue if <code>System.Web.Mvc.ValidateInputAttribute(false)</code> is set on a method.</p>
+<h2>Noncompliant Code Example</h2>
+<pre>
+public class FooBarController : Controller
+{
+    [HttpPost]
+    [ValidateInput(false)] // Noncompliant
+    public ActionResult Purchase(string input)
+    {
+        return Foo(input);
+    }
+}
+</pre>
+<h2>Compliant Solution</h2>
+<pre>
+public class FooBarController : Controller
+{
+    [HttpPost]
+    [ValidateInput(true)] // Compliant
```

Comment:
What if the user hasn't specified the attribute at all? Is it implied, or should we have a separate rule to check that the attribute is applied if there is any user input?

Reply:
Good question. This need to be covered, @agigleux shall we update the rule or add a new one?